dpnp/backend/include/dpnp_iface_random.hpp

```diff
@@ -0,0 +1,470 @@
+ *
+ * @param [in]  size   Number of elements in `result` arrays.
+ *
+ * @param [in]  l      Lot size of l.
+ *
+ * @param [in]  s      Size of sampling without replacement.
+ *
+ * @param [in]  m      Number of marked elements m.
+ *
+ * @param [out] result Output array.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void dpnp_rng_hypergeometric_c(void* result, int l, int s, int m, size_t size);
+
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (geometric distribution)
+ *
+ * @param [in]  size   Number of elements in `result` arrays.
+ *
+ * @param [in]  p      Success probability p of a trial.
+ *
+ * @param [out] result Output array.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void dpnp_rng_geometric_c(void* result, float p, size_t size);
+
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (gumbel distribution)
+ *
+ * @param [in]  size   Number of elements in `result` arrays.
+ *
+ * @param [in]  loc    The location of the mode of the distribution.
+ *
+ * @param [in]  scale  The scale parameter of the distribution.
+ *
+ * @param [out] result Output array.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void dpnp_rng_gumbel_c(void* result, double loc, double scale, size_t size);
+
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (laplace distribution)
+ *
+ * @param [in]  size   Number of elements in `result` arrays.
+ *
+ * @param [in]  loc    The position of the distribution peak.
+ *
+ * @param [in]  scale  The exponential decay.
+ *
+ * @param [out] result Output array.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void dpnp_rng_laplace_c(void* result, double loc, double scale, size_t size);
+
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (logistic distribution)
+ *
+ * @param [in]  size   Number of elements in `result` arrays.
+ *
+ * @param [in]  loc    The position of the distribution peak.
+ *
+ * @param [in]  scale  The exponential decay.
+ *
+ * @param [out] result Output array.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void dpnp_rng_logistic_c(void* result, double loc, double scale, size_t size);
+
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (lognormal distribution)
+ *
+ * @param [in]  size   Number of elements in `result` arrays.
+ *
+ * @param [in]  mean   Mean value.
+ *
+ * @param [in]  stddev Standard deviation.
+ *
+ * @param [out] result Output array.
+ *
+ */
+template <typename _DataType>
+INP_DLLEXPORT void dpnp_rng_lognormal_c(void* result, _DataType mean, _DataType stddev, size_t size);
+
+/**
+ * @ingroup BACKEND_API
+ * @brief math library implementation of random number generator (multinomial distribution)
+ *
+ * @param [in]  size          Number of elements in `result` arrays.
+ *

```

Comment:
do we need this empty line? Perhaps it is better to remove all these lines from "doctrings" in this file to make it a bit more compact.

Reply:
not sure how it will look in documentation

Reply:
@shssf I have checked, you are right. We need refactor this also, see: https://mesos.readthedocs.io/en/latest/doxygen-style-guide/.
We just need to do the same for `dpnp_iface.hpp`. Will do it on other PR.

Reply:
👍 nice article.
Please, let's do it step by step. First step inside rng only